### DISCUSSION TITLE
Option to rotate screen

### DISCUSSION THREAD

**Discussion by 5iver** [InfiniTimeOrg/InfiniTime#1620] [Ideas]
### Verification

- [X] I searched for similar feature request and found none was relevant.

### Pitch us your idea!

I'd like the screen rotated 180 degrees, so that the watch could be worn with button on the right

### Description

When I first got my PineTime and tried it on, I instinctively put it on with the button to the right, which caused the screen to be upsidedown when I turned it on! I am right-handed and have become accustomed to having buttons on the right-hand side of things and using the index finger on my right hand to toggle them. To achieve this with a PineTime, an option could be added to rotate the screen 180 degrees. Wearing the watch on the left wrist with the button on the right side would also prevent accidental button presses by sleeves, which I have encountered when wearing a winter coat or leather jacket.

**Comment by JF002** [comment]
Someone began working on it [a while ago](https://github.com/InfiniTimeOrg/InfiniTime/pull/98), but it still tagged as Work In Progress. I think we had a few issues with the hardware scrolling feature we use in InfiniTime.

**Comment by camlafit** [comment]
Hello

Could be a nice feature, as left writer, it's a bit complicated to use correctly lateral button :)
As pinetime is a square, could be interesting to manage 90° rotation

**Comment by moriel5** [comment]
Interesting, for me it feels natural to have the button facing the wrist, and as I too am left handed, that means to the left of the screen, since it is on my right hand (I am used to using my pointer/second finger to press buttons).

If anything, I would prefer to flip other watches that have the button to the right of the screen.

**Comment by vladopajic** [comment]
I am having the same discomfort when using watch as described here. Screen rotation would be vary nice feature to have.

**Comment by GDV0** [comment]
Hello,
I am working on this feature.
After 1 week, it seems ok except the following issues:
- Need to reboot the watch to apply the new orientation setting (button on the left or on the right of the screen)
- From time to time, InfinitiPaint and Paddle applications show graphic artefacts

**Comment by camlafit** [comment]
Hello

@GDV0 do you have any evolve about this feature ?

Thanks :)

**Reply by GDV0** [reply]
Still under investigation.
But, unfortunately, my sealed watch crashed and I'm not able to update the firmware anymore.

**Reply by roflbin** [reply]
Hi @GDV0,

Could you possibly share the code that you were working on? I'm sure the code, published, would be useful to someone (perhaps me, but I barely understand the codebase as it is) looking to implement this feature.

Cheers :grin:

**Comment by lmamane** [comment]
FWIIW, I had the same instinctive way to put it on… I wondered if it was designed by a left-handed person :-> Having the only button “up the arm”feels very weird.

**Comment by IngwiePhoenix** [comment]
Had just come here after asking for this here https://github.com/InfiniTimeOrg/InfiniTime/discussions/2205
...and then seeing that the PR mentioned above ( #98 ) is closed.

What are the chances this could be implemented? My right eye is blind, so I wear my watch left and would like the button to be on the right - having it on the left feels quite unnatural x)

Thank you!
